beacon-chain/rpc/service.go

```diff
@@ -21,21 +21,23 @@ var log = logrus.WithField("prefix", "rpc")
 type chainService interface {
 	IncomingBlockFeed() *event.Feed
 	IncomingAttestationFeed() *event.Feed
+	ProcessedAttestationFeed() *event.Feed
 }
 
 // Service defining an RPC server for a beacon node.
 type Service struct {
-	ctx                context.Context
-	cancel             context.CancelFunc
-	announcer          types.CanonicalEventAnnouncer
-	chainService       chainService
-	port               string
-	listener           net.Listener
-	withCert           string
-	withKey            string
-	grpcServer         *grpc.Server
-	canonicalBlockChan chan *types.Block
-	canonicalStateChan chan *types.CrystallizedState
+	ctx                   context.Context
+	cancel                context.CancelFunc
+	announcer             types.CanonicalEventAnnouncer
+	chainService          chainService
+	port                  string
+	listener              net.Listener
+	withCert              string
+	withKey               string
+	grpcServer            *grpc.Server
+	canonicalBlockChan    chan *types.Block
+	canonicalStateChan    chan *types.CrystallizedState
+	proccessedAttestation chan *pbp2p.AttestationRecord
```

Comment:
Is it possible to use `*types.Attestation`? Do we allow validator to import types package from beacon chain?

Reply:
We would have to expose the types package to the validator, either that or move it to the shared package. I dont think it is necessary to use types.Attestation as eventually when the proposer aggregates the attestations into the block he would have to do using proto.AttestationRecord , in my opinion its best if we keep it in the proto format